packages/affiliates/test/e2e/datasets.js

```diff
@@ -213,10 +186,10 @@ describe("Datasets", function() {
         end: params.startingTimestamp + 1000 * 60 * 60 * 24 * 10
       };
       const ds = Dataset(basePath, { queries, coingecko, synthPrices });
-      path = await ds.save("test-set", config);
+      path = await ds.saveDevMining("dev-mining-test", config);
```

Comment:
nit: is this test saving data to disk? My understanding was that the on-disk data was generated and put into the repo rather than being generated during the e2e test... can you familiarize me with the architecture? Or is this a test for the infrastructure that saves the on-disk data?

Reply:
This does save data on disk, but the files get removed at the end of the test. This is one of the tests that does not run in CI and needs to be run manually. It is testing the actual dataset generation utility from end to end, so that involves it saving data to disk and verifying that it can be read back out. 

The datasets that are committed were generated by the utility being tested here, and are used in the other tests that calculate reward output.

Reply:
Yeah, preferably in the readme if not already there.

Reply:
I didnt realize this but there is currently a README.md in the test/e2e folder that contains this: 

> These tests run the entire stack and reach out to external services or write to disk. The idea is that you can test as if you were in "production" as a last check to make sure everything works. These should be run locally, but maybe in the future they can be added to CI in some way.